pkg/controller/notification/notification_controller.go

```diff
@@ -433,9 +473,72 @@ func (c *Controller) syncFederatedSecret(obj *corev1.Secret) error {
 		fedObj.Spec.Template.Data = obj.Data
 		fedObj.Spec.Template.StringData = obj.StringData
 		fedObj.Spec.Template.Type = obj.Type
+	}
 
-		if err := c.Update(context.Background(), fedObj); err != nil {
-			klog.Errorf("update '%s:%s' failed, %s", fedObj.GetObjectKind().GroupVersionKind().String(), obj.Name, err)
+	err = c.updateOverrides(obj, fedObj)
+	if err != nil {
+		klog.Errorf("update FederatedSecret '%s' overrides failed, %s", obj.Name, err)
+		return err
+	}
+
+	if err := c.Update(context.Background(), fedObj); err != nil {
+		klog.Errorf("update FederatedSecret '%s' failed, %s", obj.Name, err)
+		return err
+	}
+
+	return nil
+}
+
+func (c *Controller) updateOverrides(obj *corev1.Secret, fedSecret *v1beta1.FederatedSecret) error {
+	clusterList := &v1alpha1.ClusterList{}
+	err := c.ksCache.List(context.Background(), clusterList)
+	if err != nil {
+		return err
+	}
+
+	bs, err := json.Marshal(obj.Labels)
+	if err != nil {
+		return err
+	}
+
+	fedSecret.Spec.Overrides = fedSecret.Spec.Overrides[:0]
+	for _, cluster := range clusterList.Items {
+		fedSecret.Spec.Overrides = append(fedSecret.Spec.Overrides, v1beta1.GenericOverrideItem{
+			ClusterName: cluster.Name,
+			ClusterOverrides: []v1beta1.ClusterOverride{
+				{
+					Path: "/metadata/labels",
+					Value: runtime.RawExtension{
+						Raw: bs,
+					},
+				},
+			},
+		})
+	}
+
+	return nil
+}
+
+// Update the annotations of secrets which the notification controller managed, rigger a  reconcile.
```

Comment:
=>
// Update the annotations of secrets managed by the notification controller to trigger a reconcile.